core/typedefs.h

```diff
@@ -317,4 +317,35 @@ struct BuildIndexSequence<0, Is...> : IndexSequence<Is...> {};
 #define ___gd_is_defined(val) ____gd_is_defined(__GDARG_PLACEHOLDER_##val)
 #define GD_IS_DEFINED(x) ___gd_is_defined(x)
 
+template <bool B, typename T = void>
+struct _EnableIf {
```

Comment:
I'd suggest not adding the prefix as these might be used in modules, and not necessarily needed for make them internal

Reply:
Huh! I would've assumed otherwise based on how `_GlobalLock` is prefixed, but that does seem to be the only struct in this script that uses that convention (and is part of a broader `#define`, which might explain why). And while I wouldn't personally put `too` much weight in ensuring backward compatability with modules versus something like GDExtension, this is neither the place for that discussion nor a hill I'm about to die on.

Reply:
The global lock class is strictly internal and not used outside the core, so it's different

I just don't see a reason to make this internal and just keep it as is, even if it's not used elsewhere :)

Reply:
Ahhh, I wasn't considering core-specific scope, that makes sense

Reply:
That lock is unique and used once